gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,109 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT
+		FROM pg_partition pp
+		JOIN pg_partition_rule pr ON pr.paroid = pp.oid
+		WHERE

```

Comment:
The current query has the following plan
```
"Hash Join  (cost=1.01..2.08 rows=4 width=0)"
"  Output: 1"
"  Hash Cond: (pp.oid = pr.paroid)"
"  Join Filter: CASE WHEN (pp.parrelid = '16411'::oid) THEN (('2024-04-17 16:40:39.098527+03'::timestamp with time zone >= ("substring"(pg_get_expr(pr.parrangestart, pr.parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without time zone) AND ('2024-0 (...)"
"  ->  Seq Scan on pg_catalog.pg_partition pp  (cost=0.00..1.00 rows=1 width=8)"
"        Output: pp.oid, pp.parrelid"
"  ->  Hash  (cost=1.00..1.00 rows=1 width=72)"
"        Output: pr.paroid, pr.parrangestart, pr.parchildrelid, pr.parrangeend"
"        ->  Seq Scan on pg_catalog.pg_partition_rule pr  (cost=0.00..1.00 rows=1 width=72)"
"              Output: pr.paroid, pr.parrangestart, pr.parchildrelid, pr.parrangeend"
"Optimizer: Postgres query optimizer"
```

If we add `pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass` to the WHERE section
```
SELECT 1
FROM pg_partition pp
JOIN pg_partition_rule pr ON pr.paroid = pp.oid
WHERE pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
  AND CASE
		WHEN pp.parrelid = 'arenadata_toolkit.db_files_history'::regclass
		THEN now() BETWEEN
			CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
				AS TIMESTAMP WITHOUT TIME ZONE)
			AND
			CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
				AS TIMESTAMP WITHOUT TIME ZONE)
		ELSE FALSE
	END
```
the plan become more optimal: a filter is added to Seq Scan on pg_catalog.pg_partition. 
```
"Hash Join  (cost=1.01..2.08 rows=4 width=0)"
"  Output: 1"
"  Hash Cond: (pp.oid = pr.paroid)"
"  Join Filter: CASE WHEN (pp.parrelid = '16411'::oid) THEN (('2024-04-17 16:38:13.767699+03'::timestamp with time zone >= ("substring"(pg_get_expr(pr.parrangestart, pr.parchildrelid), '#"%#"::%'::text, '#'::text))::timestamp without time zone) AND ('2024-0 (...)"
"  ->  Seq Scan on pg_catalog.pg_partition pp  (cost=0.00..1.00 rows=1 width=8)"
"        Output: pp.oid, pp.parrelid"
"        Filter: (pp.parrelid = '16411'::oid)"
"  ->  Hash  (cost=1.00..1.00 rows=1 width=72)"
"        Output: pr.paroid, pr.parrangestart, pr.parchildrelid, pr.parrangeend"
"        ->  Seq Scan on pg_catalog.pg_partition_rule pr  (cost=0.00..1.00 rows=1 width=72)"
"              Output: pr.paroid, pr.parrangestart, pr.parchildrelid, pr.parrangeend"
"Optimizer: Postgres query optimizer"
```
When pg_partition has many tuples, planner can use the pg_partition_parrelid_index index.